### ISSUE TITLE
How to deal with the case protobuf does not provide enough data for the color tables?

### ISSUE THREAD

**Issue by viniul** [viniul/oss-fuzz#4]
@bshastry
Consider: 
```
void ProtoConverter::visit(GlobalColorTable const& gct)
{
	// This has to contain exactly 3*2^(m_GlobalColorExp + 1) bytes
        m_output.write(gct.colors().data(), gct.colors().size());
}
```
How do we handle the case where `3*2^(m_GlobalColorExp + 1) > gct.colors.size()`?

We could:
1. Bail out -> That would waste a lot of test cases.
2. Just write the whole gct.colors.data(), ignore the rest of the data.

Is there maybe a better option?

**Comment by bshastry** [comment]
Yeah, I'm not sure what's the ideal solution here. Of the three cases
  - gct.colors.size() > 3*2^(m_GlobalColorExp + 1)
  - gct.colors.size() < 3*2^(m_GlobalColorExp + 1)
  - gct.colors.size() == 3*2^(m_GlobalColorExp + 1)

I feel that the second is probably more likely, followed by the first and third. I was hoping we could somehow make it roughly 33% probability each. Not sure how, may be we should file an issue over at protobuf to ask how to handle such dependencies.

**Comment by viniul** [comment]
That makes sense. Also we need to evaluate the trade-off "adhering to the standard" vs. "creating interesting edge-cases". So if we always ensure that the Global Color Table has the correct size, then we might miss some edge cases?

**Comment by bshastry** [comment]
Yeah, I was wondering if a third of the time getting it right and wrong rest of the time is good enough, but I am pretty sure if that's not ideal. Still, currently we should be getting it right < 1% so perhaps that can be made better
